components/event-publisher-proxy/pkg/sender/nats.go

```diff
@@ -41,16 +41,13 @@ func (h *NatsMessageSender) Send(ctx context.Context, event *cev2event.Event) (i
 		return http.StatusInternalServerError, err
 	}
 
-	ctxWithCancel, cancel := context.WithCancel(ctx)
-	defer cancel()
-
 	c, err := cev2.NewClient(p)
 	if err != nil {
 		h.Logger.Errorf("Failed to create client, %s", err.Error())
 		return http.StatusInternalServerError, err
 	}
 
-	err = c.Send(ctxWithCancel, *event)
+	err = c.Send(ctx, *event)
```

Comment:
Why are we removing the context with cancel?

Reply:
ctx is already a "cancelable" context.